File: solutions/dataStructures/arrays/bestTimeToBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        max_profit = 0
        min_buy = prices[0]

        for i in range(1, n):
            max_profit = max(max_profit, prices[i] - min_buy)
            min_buy = min(min_buy, prices[i])
        
        return max_profit

File: solutions/java/grind75/week1/bestTimeToBuyAndSellStock.java
class Solution {
    public int bestTimeToBuyAndSellStock(int[] prices) {
        int n = len(prices);
        int max_profit = 0
        int min_buy = prices[0]

        for (int i = 1; i < prices.length; i++) {
            max_profit = max(max_profit, prices[i] - min_buy)
            min_buy = min(min_buy, prices[i])
        }
            
        return max_profit
}